vendor.go

```diff
@@ -191,7 +206,7 @@ func addCleanup(c func()) {
 
 func abortonerr(err error, details string) {
 	if err != nil {
-		fmt.Sprintf("unexpected error[%s] %s\n", err, details)
+		fmt.Printf("unexpected error[%s] %s\n", err, details)
 		cleanup()
```

Comment:
AHHHHH YEAHHH that is why this whole cleanup crap...it is to guarantee that stuff is cleaned up before calling os.Exit. Today I'm more prone to just panicking instead of calling os.Exit because it will call all defers, ensuring everything is cleaned up. (ensure is kinda a strong word, but I imagine you get the idea =P). I don't know why I was all against panics, today I'm much more okay with it, specially as an alternative from calling os.Exit (but not panics in libs, panics in libs is still pretty bad for me =P).

Reply:
The idea of cleanup funcs makes sense for garbage like temporary files and directories (as is the case) but not closing open files and so, because of system limits.
But no worry man, I know you did in a rush.

Reply:
But usually defer is simpler, this was just stupid